client/src/components/MyMap.js

```diff
@@ -12,6 +12,28 @@ function MyMap() {
 	const [isModalOpen, setModalOpen] = useState(false);
 	const [selectedRhythm, setSelectedRhythm] = useState({});
 
+	function toggleDropdownShown(name) {
+		setRhythmCodes((prevRhythmCodes) => {
+			const newRhythmCodes = [];
+			for (let i = 0; i < prevRhythmCodes.length; i++) {
```

Comment:
This is a clever solution to the problem and it works, but it's not very obvious what's happening in this code. It would be good to add a comment explaining to your fellow developers.

Reply:
Thank you, notes added  as suggested 👍